packages/contracts/contracts/src/AdvancedChecker.sol

```diff
@@ -26,7 +29,7 @@ abstract contract AdvancedChecker is IAdvancedChecker {
     /// @param evidence Validation data.
     /// @param checkType Check type to perform.
     /// @return checked Validation result.
-    function _check(address subject, bytes memory evidence, Check checkType) internal view returns (bool checked) {
+    function _check(address subject, bytes[] calldata evidence, Check checkType) internal view returns (bool checked) {
```

Comment:
any reason why we move from memory to calldata? don't we need to allow the flexibility for devs to enforce logic that might change some of the evidence (for whatever reason)?

Reply:
I think this should be never the case since can invalidate the evidence. There could potentially be use cases for modifying the evidence? If yes, they can just copy the piece of data they need from the evidence and apply their logic on top.

Reply:
Like, this maintains the principle of "evidence as proof" while still giving developers flexibility to transform data when needed, just not on the evidence itself. Wdyt?

Reply:
> "evidence as proof" 

yea makes sense